### PR TITLE
diff: add sql-mode config

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
some database doesn't use default SQL mode, and the diff will parse failed when getting table struct. 

### What is changed and how it works?
add sql-mode config for database config

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Related changes

 - Need to update the documentation

### CODE REVIEW

**Review by lichunzhu — COMMENTED** [review]
LGTM

File: pkg/dbutil/table_test.go [hunk 1]
```diff
@@ -95,15 +95,15 @@ func (*testDBSuite) TestTable(c *C) {
...
 	c.Assert(err, IsNil)
 
 	createTableSQL2 := "CREATE TABLE `test`.`atest` (`id` int(24) NOT NULL, `name` varchar(24), `birthday` datetime, `update_time` time, `money` decimal(20,2), primary key(`id`))"
-	tableInfo2, err := GetTableInfoBySQL(createTableSQL2)
+	tableInfo2, err := GetTableInfoBySQL(createTableSQL2, "")
 	c.Assert(err, IsNil)
 
-	createTableSQL3 := "CREATE TABLE `test`.`atest` (`id` int(24), `name` varchar(24), `birthday` datetime, `update_time` time, `money` decimal(20,2), unique key(`id`))"
-	tableInfo3, err := GetTableInfoBySQL(createTableSQL3)
+	createTableSQL3 := "CREATE TABLE \"test\".\"atest\" (\"id\" int(24), \"name\" varchar(24), \"birthday\" datetime, \"update_time\" time, \"money\" decimal(20,2), unique key(\"id\"))"
```
Comment (csuzhangxc) [inline pkg/dbutil/table_test.go]: you can use ``` ` ``` to wrap literal string.

File: tests/sync_diff_inspector/snapshot/run.sh [hunk 2]
```diff
@@ -21,8 +21,11 @@ check_contains "sourceDB don't equal targetDB" $OUT_DIR/snapshot_diff.log
 cp $OUT_DIR/fix.sql $OUT_DIR/fix.sql.bak
 
 echo "use snapshot compare data, data should be equal"
+# test sql mode by the way
+mysql -uroot -h 127.0.0.1 -P 4000 -e "SET GLOBAL sql_mode = 'ANSI_QUOTES';"
 cp config_base.toml config.toml
 echo "snapshot = \"$ts\"" >> config.toml
+echo "sql-mode = 'ANSI_QUOTES'" >> config.toml
```
Comment (csuzhangxc) [inline tests/sync_diff_inspector/snapshot/run.sh:34]: If remove this line, can the test pass?

File: tests/sync_diff_inspector/snapshot/run.sh [hunk 3]
```diff
@@ -20,15 +20,25 @@ check_contains "sourceDB don't equal targetDB" $OUT_DIR/snapshot_diff.log
...
-echo "use snapshot compare data, data should be equal"
+echo "use snapshot compare data, test sql mode by the way, will return error, diff should not passed"
+mysql -uroot -h 127.0.0.1 -P 4000 -e "SET GLOBAL sql_mode = 'ANSI_QUOTES';"
 cp config_base.toml config.toml
 echo "snapshot = \"$ts\"" >> config.toml
 sync_diff_inspector --config=./config.toml > $OUT_DIR/snapshot_diff.log
+check_contains "sourceDB don't equal targetDB" $OUT_DIR/snapshot_diff.log
+
+echo "use snapshot compare data, data should be equal"
+mysql -uroot -h 127.0.0.1 -P 4000 -e "SET GLOBAL sql_mode = 'ANSI_QUOTES';"
```
Comment (csuzhangxc) [inline tests/sync_diff_inspector/snapshot/run.sh]: duplicate with L24?

File: tests/sync_diff_inspector/snapshot/run.sh [hunk 4]
```diff
@@ -20,15 +20,25 @@ check_contains "sourceDB don't equal targetDB" $OUT_DIR/snapshot_diff.log
...
+sync_diff_inspector --config=./config.toml > $OUT_DIR/snapshot_diff.log
 check_contains "test pass!!!" $OUT_DIR/snapshot_diff.log
 
 echo "execute fix.sql and use base config, and then compare data, data should be equal"
 cat $OUT_DIR/fix.sql.bak | mysql -uroot -h127.0.0.1 -P 4000
 sync_diff_inspector --config=./config_base.toml > $OUT_DIR/snapshot_diff.log
 check_contains "test pass!!!" $OUT_DIR/snapshot_diff.log
 
+# reset sql mode
+mysql -uroot -h 127.0.0.1 -P 4000 -e "SET GLOBAL sql_mode = 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';"
```
Comment (csuzhangxc) [inline tests/sync_diff_inspector/snapshot/run.sh:45]: how about query this from the target database before `SET GLOBAL sql_mode = 'ANSI_QUOTES'`?

File: tests/sync_diff_inspector/snapshot/run.sh (hunk 2 above)
Comment (WangXiangUSTC) [reply tests/sync_diff_inspector/snapshot/run.sh:34]: add a test in 47562ca

File: tests/sync_diff_inspector/snapshot/run.sh (hunk 3 above)
Comment (WangXiangUSTC) [reply tests/sync_diff_inspector/snapshot/run.sh]: removed

File: tests/sync_diff_inspector/snapshot/run.sh (hunk 4 above)
Comment (WangXiangUSTC) [reply tests/sync_diff_inspector/snapshot/run.sh:45]: it is hard to get sql mode from shell, how about just use these default sql mode

Comment (csuzhangxc) [reply tests/sync_diff_inspector/snapshot/run.sh:45]: fine. it's not a problem.

**Review by csuzhangxc — DISMISSED** [review]
LGTM

File: tests/sync_diff_inspector/snapshot/run.sh [hunk 5]
```diff
@@ -20,15 +20,28 @@ check_contains "sourceDB don't equal targetDB" $OUT_DIR/snapshot_diff.log
 # fix.sql will be empty after check below, so backup it
 cp $OUT_DIR/fix.sql $OUT_DIR/fix.sql.bak
 
-echo "use snapshot compare data, data should be equal"
+echo "use snapshot compare data, test sql mode by the way, will return error, diff should not passed"
+mysql -uroot -h 127.0.0.1 -P 4000 -e "SET GLOBAL sql_mode = 'ANSI_QUOTES';"
+sleep 10
+mysql -uroot -h 127.0.0.1 -P 4000 -e "show variables like '%sql_mode%'"
+mysql -uroot -h 127.0.0.1 -P 4000 -e "show create table diff_test.test"
```
Comment (csuzhangxc) [inline tests/sync_diff_inspector/snapshot/run.sh:27]: why do we need these lines?

File: tests/sync_diff_inspector/snapshot/run.sh [hunk 6]
```diff
@@ -20,15 +20,28 @@ check_contains "sourceDB don't equal targetDB" $OUT_DIR/snapshot_diff.log
...
 cp config_base.toml config.toml
 echo "snapshot = \"$ts\"" >> config.toml
+sync_diff_inspector --config=./config.toml > $OUT_DIR/snapshot_diff.log || true
```
Comment (csuzhangxc) [inline tests/sync_diff_inspector/snapshot/run.sh:30]: Is `|| true` necessary?

File: tests/sync_diff_inspector/snapshot/run.sh (hunk 5 above)
Comment (WangXiangUSTC) [reply tests/sync_diff_inspector/snapshot/run.sh:27]: just for debug, set sql_mode will not take effect immediately in tidb, so need to wait some time and then check it

File: tests/sync_diff_inspector/snapshot/run.sh (hunk 6 above)
Comment (WangXiangUSTC) [reply tests/sync_diff_inspector/snapshot/run.sh:30]: yes, diff will get error, and this script will exit

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**